apps/teams-test-app/src/components/Links.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { ReactElement } from 'react';
+
+import { ApiWithTextInput } from './utils';
+
+const MoveToLink = (): ReactElement =>
+  ApiWithTextInput<string>({
+    name: 'moveToLink',
+    title: 'Move to Link',
+    onClick: {
+      validateInput: input => {
+        if (typeof input !== 'string') {
+          throw new Error('Input should be a string');
+        }
+      },
+      submit: async input => {
+        window.location.href = input;
+        return Promise.resolve('Moved to new link');
+      },
+    },
+  });
+
+const OpenLinkInNewWindow = (): ReactElement =>
+  ApiWithTextInput<string>({
+    name: 'openLinkInNewWindow',
+    title: 'Open link in new window',
+    onClick: {
+      validateInput: input => {
+        if (typeof input !== 'string') {
+          throw new Error('Input should be a string');
+        }
+      },
+      submit: async input => {
+        window.open(input);
+        return Promise.resolve('Link opened');
+      },
+    },
+  });
+
+const Links = (): ReactElement => (
+  <>
+    <h1>Links</h1>
+    <a id="link_simple" href="https://teams-test-tab.azurewebsites.net/">
```

Comment:
>https://teams-test-tab.azurewebsites.net/ [](http://example.com/codeflow?start=30&length=41)

Are you using this URL because it should point to the current app? Or just because you know it points to a valid instance of the test app? I worry about hardcoding a URL like this into the test app.

Reply:
I was trying to point this to a URL which is different from test app but is still a URL which can communicate with the Hub. This would help us validate the redirect scenarios i.e. when an App redirect to a different valid URL, it should be able to use the APIs